plugins/themes/default/js/lib/bootstrap/carousel.js

```diff
@@ -0,0 +1,595 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v4.6.2): carousel.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE)
```

Comment:
I'm pulling the carousel from `stable-3_4_0`. I'll just include it in the `main` PRs.

I'll see if it's possible to pull in a fully-built version through a dependency manager. But it may not be very easy. Most modules in the npm ecosystem expect to be imported and run through a compile (ie - `npm run build`).

Reply:
Ok, looking at this further, I'm not sure what's the best way to pull in the dependency. If I use [swiper.js](https://swiperjs.com/), which looks like a good option, I can get at prebuilt assets in the `node_modules` directory by loading them in the theme:

```php
$this->addScript('swiper', $request->getBaseUrl() . '/node_modules/swiper/swiper-bundle.js', ['baseUrl' => '']);
$this->addStyle('swiper', $request->getBaseUrl() . '/node_modules/swiper/swiper-bundle.css', ['baseUrl' => '']);
```

Of course, those files only exist in a development context. I could add a step to `buildpkg.sh` that copies those files over into the theme directory. However, then these won't be available in a development context. I could work around this by making the default theme load the dependencies from `node_modules` if available and from the theme if not. But that feels like getting into dev/prod environments, which we don't really have...

Any ideas on how you want to proceed with this?

Reply:
@jardakotesovec, any suggestions on this? Big picture: we're trying to decrease the number of 3rd-party bits of code that are dumped into our github repo without some kind of management.

Reply:
@jardakotesovec, I'd still love any suggestions you might have on this!

@NateWr, we did have a similar discussion about a file that's part of a Composer dependency, and I think we decided to use Composer to copy it somewhere outside of the `lib/pkp/lib/vendor` directory where we can rely on it being present. (This applies to both dev and build, and the file gets put in place during Composer dependency installation.)

The advantage is that we'll get clear breakage during dependency install if the file moves or changes name inside the package.

Something like [this Stackoverflow discussion](https://stackoverflow.com/questions/48777872/copy-files-from-package-to-project-root-directory-using-composer). It looks like there's a [similar need in NPM](https://stackoverflow.com/questions/38026117/how-to-use-package-json-scripts-to-copy-files-with-specific-file-extension).

Reply:
Ok, I've moved swiper to a npm depdency in package.json, and added a `postinstall` script to copy the required files into the default theme: https://github.com/pkp/ojs/pull/4038/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14

If the tests pass for OJS, I'll roll out that change to the OMP/OPS PRs.

Reply:
Sorry for delay. We had similar situation with pflPlugin (https://github.com/asmecher/pflPlugin/blob/main/package.json), where I end up using package.json and simple vite build step to use these node_modules. 

In any case having it as npm dependency is what we want (we don't want to introduce some additional mechanism). 

I am not sure how much hassle introducing own package.json to default theme plugin would cause - it would require additional `npm install` and potentially `npm run build` to get things ready.

My guess that overtime we will introduce own package.json for default theme and add some postinstall steps to the top package.json to install dependencies for the default theme. But I suspect thats out of the scope of this effort, so  adding swiper.js to the top package.json and making copy looks as good compromise for now.